### ISSUE TITLE
RemoteRScriptRepository and GitHubScriptRepository

### ISSUE THREAD

**Issue by nuest** [52North/WPS#80]
Implement an abstract class `RemoteRScriptRepository` that can manage scripts from remote locations, namely URLs or URIs. On startup scripts are copied into a local directory (based on an an abstract method `public abstract Collection<InputStream> getScripts()` and then loaded. The loading functions can probably be re-used from the `LocalR...Repository`. The remote repository should allow to set a fixed interval on which it updates the local copies of the scripts.

The `GitHubScriptRepository` will allow to define a GitHub username and repository name and from that information download the R files from that repository.
- JGit is EDL (http://eclipse.org/jgit/ - https://wiki.52north.org/bin/view/Documentation/ThirdPartyLicensesForGPL) and on Maven Central (http://mvnrepository.com/artifact/org.eclipse.jgit/org.eclipse.jgit) so we can just clone the repo!
- Maybe calling `git clone` on the command line via `JNDI` is a more lightweight alternative, but it requires git to be installed.
- Or the GitHub API (https://developer.github.com/v3/repos/#get), e.g. https://api.github.com/repos/52North/WPS - but I didn't find an option for file download...

**Comment by nuest** [comment]
By now I would prefer using JGit from the start, it allows to also update the local clone outside of the WPS, for example.

Related and maybe also interesting: ``rgist`` package to access gists from R.

**Comment by matthias-mueller** [comment]
Okay, let's see where this goes. I see four or five things that we have previously discussed within the repository concept:

1) Access to executable workspaces or code
2) Access to related source code (especially important for byte code)
3) Versioning
4) Documentation
5) Link with github due to its widespread use

I'd be glad if we could keep things as simple as possible, i.e. not force our users into complex solutions if they just want to upload a simple script. Although I have a strong affinity to using GIT, we should still be able to handle plain (non-git) workspaces and content.

**Comment by nuest** [comment]
Yes - this is an addition to the simple "drop-in folder" solution that WPS4R already supports.

If you would like to start a discussion on the repository concept in general, could you open a new issue?
